src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala

```diff
@@ -39,7 +39,11 @@ import java.time.{Duration, Period}
   */
 class FeelInterpreter {
 
-  def eval(expression: Exp)(implicit context: EvalContext): Val =
+  def eval(expression: Exp)(implicit context: EvalContext): Val = {
+    if (Thread.interrupted()) {
+      throw new InterruptedException()
+    }
```

Comment:
:wrench: We should add a comment to explain why we are checking for the interruption. 

It adds context for the next devs who is looking at this code.